### PR TITLE
@ws docs

### PR DESCRIPTION
**Pull request by Ankcorn**
### We really appreciate your commitment to improving Architect

To maintain a high standard of quality in our releases, before merging every pull request we ask that you've completed the following:

- [x] Forked the repo and created your branch from `master`
- [x] Made sure tests pass (run `npm it` from the repo root)
- [x] Updated relevant documentation internal to this repo (e.g. `readme.md`, help docs, inline docs & comments, etc.)
- [x] Linked to any related issues, PRs, etc. below that may relate to, consume, or necessitate these changes

Please also be sure to completed the CLA as it's required for your PR to be merged. A github comment will prompt you if you haven't already.

Learn more about [contributing to Architect here](https://arc.codes/intro/community).

Thanks again!

Let me know if you have any suggestions for improvements on the docs

I assume this https://github.com/architect/architect/issues/1099 will be fixed but could add a note about a workaround if that is easier.

### CODE REVIEW

File: src/views/docs/en/reference/app.arc/ws.md [hunk 1]
```diff
@@ -81,28 +68,48 @@ http=[
...
+│   ├── connect
+│   ├── default
+│   ├── disconnect
+├── app.arc
+└── package.json
 ```
+
+Each function responds to websocket events from a client.
+
+In the functions payload there is a connectionId that uniquely identifies a client. This connectionId can be used to send messages to the correct client.
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:94]:
```suggestion
In the [payload delivered to the function](#function-payload) there is a `connectionId` that uniquely identifies a client. This `connectionId` can be used to send messages to the correct client.
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 2]
```diff
@@ -81,28 +68,48 @@ http=[
...
-'-app.arc
+├── ws
+│   ├── connect
+│   ├── default
+│   ├── disconnect
+├── app.arc
+└── package.json
 ```
+
+Each function responds to websocket events from a client.
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:92]:
```suggestion
Each function responds to WebSocket events from a client.
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 3]
```diff
@@ -81,28 +68,48 @@ http=[
...
+
+In the functions payload there is a connectionId that uniquely identifies a client. This connectionId can be used to send messages to the correct client.
+
+### Functions
+
+The functions created by the @ws pragma handle events from a websocket client.
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:98]:
```suggestion
The functions created by the `@ws` pragma handle events from a WebSocket client.
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 4]
```diff
@@ -81,28 +68,48 @@ http=[
...
+
+* Connect - This function is invoked when a websocket client connects to the application
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:100]:
```suggestion
* Connect - This function is invoked when a WebSocket client connects to the application
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 5]
```diff
@@ -81,28 +68,48 @@ http=[
...
+* Default - This function is invoked when a websocket client sends a message to the application
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:101]:
```suggestion
* Default - This function is invoked when a WebSocket client sends a message to the application
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 6]
```diff
@@ -81,28 +68,48 @@ http=[
...
+* Disconnect - This function is invoked when a websocket client disconnects from your application
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:102]:
```suggestion
* Disconnect - This function is invoked when a WebSocket client disconnects from your application
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 7]
```diff
@@ -81,28 +68,48 @@ http=[
...
+
+#### Function Payload
+
+|Argument|Description|
+|---|---|
+|`req`|The websocket request payload|
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:108]:
```suggestion
|`req`|The WebSocket request payload|
```

File: src/views/docs/en/reference/app.arc/ws.md [hunk 8]
```diff
@@ -81,28 +68,48 @@ http=[
...
+|`req.requestContext.connectionId`|An id that uniquely identifies a client|
+
+#### Send Messages
+
+To publish a message to a websocket client you can use the architect/functions ws.send method. You can call this method from any of your applications functions.
```
Comment (filmaj) [inline src/views/docs/en/reference/app.arc/ws.md:113]:
```suggestion
To publish a message to a WebSocket client you can use arc's runtime library `@architect/functions`' `ws.send` method. You can call this method from any of your application's functions.
```

File: src/views/docs/en/reference/runtime/node.md [hunk 9]
```diff
@@ -31,6 +31,7 @@ let arc = require('@architect/functions')
 - [`arc.tables`](#arc.tables) Generates a DynamoDB client for `@tables`
 - [`arc.events`](#arc.events) Publish/subscribe helpers for SNS `@events` functions
 - [`arc.queues`](#arc.queues) Publish/subscribe helpers for SQS `@queues` functions
+- [`arc.ws`](#arc.ws) Websocket helpers for `@ws` functions
```
Comment (filmaj) [inline src/views/docs/en/reference/runtime/node.md:34]:
```suggestion
- [`arc.ws`](#arc.ws) WebSocket helpers for `@ws` functions
```

File: src/views/docs/en/reference/runtime/node.md [hunk 10]
```diff
@@ -233,3 +234,16 @@ await arc.queues.publish({
   payload: {hits: 1},
 })
 ```
+
+### `arc.ws`
+
+Send a message over websockets
```
Comment (filmaj) [inline src/views/docs/en/reference/runtime/node.md:240]:
```suggestion
Send a message over WebSockets.
```

**Review by filmaj — APPROVED** [review]
This looks great, thanks for doing this! I left a few suggested edits (mostly formatting). Looking forward to merging this in, cheers!

**Comment by Ankcorn** [comment]
Thanks for all the suggested changes!

Happy to help 🤗

**Comment by filmaj** [comment]
Sweet, merged and tagged! Should be online in a few minutes.

**Pull request merged**